### DISCUSSION TITLE
Matching gif duration on a watermark

### DISCUSSION THREAD

**Discussion by marcb1387** [ImageMagick/ImageMagick#2557] [Help]
i am watermarking a gif on top of another gif but the duration for the watermark are not long enough. is it possible to loop the watermark to the duration of the gif ? 

i have currently used this older thread to be able to batch process the images but have run into this issue of the duration not being the same. http://www.imagemagick.org/discourse-server/viewtopic.php?t=25731

**Comment by fmw42** [comment]
What is your command line?  What is your ImageMagick version and platform/OS.  Do you want to reverse the animation or just duplicated it so that it is longer?  See -duplicate at https://imagemagick.org/Usage/basics/#duplicate or -reverse at https://imagemagick.org/Usage/basics/#reverse

**Comment by marcb1387** [comment]
Here is my current code(code from previous thread) that will place the watermark in but the watermark need to be duplicated to match the same duration and the input gif. On windows, 7.0.10

```
set INDIR=tests
set OUTDIR=fin

for /F "usebackq" %%F in (`dir /b %INDIR%\*.gif`) do (
  convert ^
    %INDIR%\%%F ^
    -coalesce ^
    -gravity SouthWest -geometry +0+0 null: test.gif ^
    -layers composite -layers optimize ^
    %OUTDIR%\%%~nxF
)
```

**Reply by fmw42** [reply]
I understand the code, but not what you want to do to change the animation.

**Comment by marcb1387** [comment]
the images that i need to put the watermark on are different duration's. i'm thinking i need to read the duration of each input gif and then adjust the watermark gif (test.gif) by duplicate to make it the proper duration of the input gif.

one of my input gif duration's are 1770MS long. The watermark gif is 960MS and i need to extend that duration by duplicate i suppose to reach the 1770MS duration length, But i wanted to see if i can do this programmatically since my input gif's have varying duration's.

**Comment by snibgo** [comment]
You have two input GIFs. If you need the result to contain an integer number of loops of both inputs, then the number of frames in the output should be a common multiple of the frames in both inputs.

For example, if your inputs have 300 frames and 400 frames, then the output could have 1200 frames.

However, if the frame rates (frames per second) of the inputs are different, then you first need to temporally resample one or both inputs. The best tool for that is ffmpeg.

EDIT: There are alternatives. For example, you could repeat the last frame of the 300 frame GIF to make it 400 frames long.
